lib/sdf-server/src/service/v2/audit_log/list_audit_logs.rs

```diff
@@ -27,15 +34,128 @@ pub async fn list_audit_logs(
     AccessBuilder(access_builder): AccessBuilder,
     Path((_workspace_pk, change_set_id)): Path<(dal::WorkspacePk, dal::ChangeSetId)>,
     Query(request): Query<ListAuditLogsRequest>,
+    State(state): State<AppState>,
 ) -> AuditLogResult<Json<ListAuditLogsResponse>> {
     let ctx = builder
         .build(access_builder.build(change_set_id.into()))
         .await?;
 
-    let (logs, can_load_more) = audit_logging::list(&ctx, request.size.unwrap_or(0)).await?;
+    let (database_logs, can_load_more) = audit_logging::list(
+        &ctx,
+        state.audit_database_context(),
+        request.size.unwrap_or(0),
+    )
+    .await?;
+
+    let mut assembler = Assembler::new();
+    let mut logs = Vec::with_capacity(database_logs.len());
+    for database_log in database_logs {
+        logs.push(assembler.assemble(&ctx, database_log).await?);
+    }
 
     Ok(Json(ListAuditLogsResponse {
         logs,
         can_load_more,
     }))
 }
+
+#[derive(Debug)]
+struct Assembler {
+    change_set_cache: HashMap<ChangeSetId, ChangeSet>,
+    user_cache: HashMap<UserPk, User>,
+}
```

Comment:
This was ported from the prior implemented with the NATS JetStream stream. We may want the metadata that these caches provide for the frontend in the database in its own columns, but we decided to just extend the metadata on the way out to `app/web` for now.

cc @britmyerss @fnichol since we talked about this